shared/src/main/scala/pl/metastack/metatime/Date.scala

```diff
@@ -106,8 +106,9 @@ object Date {
     }
   }
 
-  def findMonth(days: Int, year: Int): Int =
-    accumulate(accuDays, year).filter(x => x <= days).length + 1
+  def findMonth(days: Int, year: Int): Int = {
+    accumulate(accuDays, year).filter(x => x <= math.abs(days)).length + 1
```

Comment:
You should not pass a negative value in the first place when calling `findMonth()`. You could add `require(days >= 0 && year >= 0)` to verify the value ranges.